markup/asciidoc/convert.go

```diff
@@ -64,7 +65,8 @@ func (a *asciidocConverter) getAsciidocContent(src []byte, ctx converter.Documen
 	}
 
 	a.cfg.Logger.INFO.Println("Rendering", ctx.DocumentName, "with", path, "...")
-	args := []string{"--no-header-footer", "--safe"}
+	folderName := filepath.Dir(filepath.Join(a.cfg.Cfg.GetString("contentDir"), ctx.DocumentName))
```

Comment:
There isn't ONE config dir. We could probably add `Filename` (full path) to the `DocumentContext` (but should also then rename the existing attributes to reflect what they contain; `DocumentName` => Path).

Reply:
The working dir problem is not just to get asciidoc `include::somefile.adoc[]` working. Also `asciidoctor-diagram` rendered images in the wrong folder. 

My proposal to solve that is exposing all asciidoctor args to `config.toml` and adding `--base-dir` and `outdir=` to asciidoctor args if desired. 

So this pull request includes:
 
1. New configuration parameters for `config.toml` e.g.:
```
asciidoctorArgs = ["--no-header-footer", "-r", "asciidoctor-html5s", "-b", "html5s", "-r", "asciidoctor-diagram"]
asciidoctorCurrentContent = true
```

2. Unit tests that prove that these changes can be configured, so we stay backward compatible (changing the working dir is a breaking change!)

3. I tested that in a sample repo https://github.com/muenchhausen/hugo_asciidoc_ddd

What is open? 
- I am not sure if the new configuration parameters follow the hugo naming convention - please check! 
- Please let me know, where I can document the new configuration options.

Reply:
>I am not sure if I fully understood @bep `s comment - I will try to dig deeper

I suspect that, too. I will try to explain better.

* Hugo can have many content dirs (while the common thing is to have 1). Hugo Modules means that this can be "virtual dirs" on GitHub.
* Given the above, the only current way to get the real filename to a content file is to look at the `Filename`. We have a `DocumentContext` with some badly named fields, but `DocumentName`I think is filled with `Path`; this is the file's path relative to the content dir.  So, to get to the real filename, I think the simplest is if you add `Filename` to the `DocumentContext` (you could probably also do a `.ContentFs.Stat`, but that is a little bit ugly).

A final question: I don't use Asciidoctor myself, and have not invested too much time in it: If we add some flags we must make sure that they don't break if run at a reasonably recent version of both asciidoc and asciidoctor... So, if the above flags is for asciidocor, but not defined in asciidoc ...

Reply:
Thanks for explaining! I think adding `FileName` to `DocumentContext` is the correct proposal for converters that need to reference files. This works with modules very well.

I wrote some tests to ensure that `asciidoc` still works. Anyhow, the new features just covers `asciidoctor` because I have just experience with that.

The naming of the configuration parameters are improved now.

Open: Documentation. Should we write a dedicated chapter on `asciidoc`? What are your thoughts?

Reply:
AsciiDoc EOLs in Jan. I suggest to depecated this for Hugo from now on.

I renamed FileName to Filename. 

I added documentation today - please have a look if this is fine!

Reply:
@bep I would like to deploy this patch to 3 installations this week, test it and continue on some new improvements. Is there anything open I have to improve?